### ISSUE TITLE
Stubbing method with return type of Future<void> is not possible with null safety

### ISSUE THREAD

**Issue by gysipos** [dart-lang/mockito#381]
Hi!

My Mock object (of Hives Box) has a delete method witch returns Future of type void and i simply cannot stub it.

Here is my annotation:
`@GenerateMocks([], customMocks: [MockSpec<Box<CachedLocation>>()])
` 
but i also tried:
`@GenerateMocks([], customMocks: [MockSpec<Box<CachedLocation>>(returnNullOnMissingStub: true)])
`

here is how i try to stub:
  when(box.delete(any)).thenAnswer((_) async {});

also tried these versions:
```
  Future<void> returnVoid() async {}
  when(box.delete(any)).thenAnswer((_) async => returnVoid());
```
```
when(box.delete(any)).thenAnswer((_) {
  Future<void> result = Future.value();
  return result;
});

```
```
Future<void> iWasntAsking = Future.value();
when(box.delete(any)).thenReturn(iWasntAsking);
```
`when(box.delete(any)).thenAnswer((_) async => Future.value()));`

I get the same result in every configuration of the above options, even if i don't stub at all and the  `returnNullOnMissingStub` is true:

The test fails in the line of the stub with the error message:
`type 'Null' is not a subtype of type 'Future<void>'`

The only other answer i can get if i don't stub and i set `returnNullOnMissingStub` to false. In that case it complains about no stub found for the method.

So, i really don't know how should i stub a method that returns Future<void> in a null-safe environment.

i'm on `  mockito: ^5.0.3
`

**Comment by srawlins** [comment]
I believe the newly published mockito 5.0.4 should solve this issue, and you won't need `returnNullOnMissingStub: true`.

**Comment by nakajo2011** [comment]
I am using v5.0.5, but not solved this issue.

**Comment by srawlins** [comment]
Hi @nakajo2011 can you explain what issue you are having? Can you show your code for reference?

**Comment by FirentisTFW** [comment]
I have the same issue and I'm using v5.0.5.
I just did a couple of tests and it doesn't seem to work even on functions with non-Future return types.

I tried several options, here's what I found:
1. If you override method directly in mocked class and not stub it in test, it works.
2. If you don't override method in mocked class and you stub it in test, it shows error message:
`type 'Null' is not a subtype of type 'Future<String>'`
3. For me, it doesn't seem to work on any type, not only Futures. So not only thenAnswer doesn't work, but thenReturn seems not to be working either.
4. It works fine with older SDK versions without null safety.
